bot/exts/moderation/infraction/_utils.py

```diff
@@ -298,3 +300,23 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    @discord.ui.button(label="Confirm", style=ButtonStyle.red)
+    async def confirm(self, interaction: Interaction, button: Button) -> None:
+        """Callback coroutine that is called when the "confirm" button is pressed."""
+        await interaction.response.defer()
+
+        self.confirmed = True
```

Comment:
I think it would be nice to initialise `self.confirmed` to `False` in the `__init__`. It shouldn't make a difference with the current logic but it makes the code a bit easier to reason about as the attribute is always defined.